IoTCIntegration/lib/engine.js

```diff
@@ -116,7 +116,7 @@ async function getDeviceHub(context, device) {
         method: 'PUT',
         json: true,
         headers: { Authorization: sasToken },
-        body: { registrationId: deviceId }
+        body: { registrationId: deviceId, payload: { iotcModelId: device.modelId } }
```

Comment:
Did you check sending a message without  specifying a modelId (just to make sure that passing undefined here won't break it).

Reply:
Yep, works fine